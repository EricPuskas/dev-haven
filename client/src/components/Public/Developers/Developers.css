.Developers {
  display: flex;
  justify-content: center;
  flex-direction: column;
  background-color: #fff;
  width: 100%;
  align-items: center;
  text-align: start;
  height: 80vh;
  padding: 2rem;
  margin: 0 auto;
  margin-bottom: 5rem;
  font-family: "Ubuntu";
}

.Developer-profile {
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  padding: 1rem;
  align-items: center;
  margin-bottom: 5rem;
  font-family: "Ubuntu";
  min-width: 1000px;
}
.Developers h1 {
  color: rgb(48, 48, 48);
}

.Developers p {
  font-size: 1.1rem;
}

.Developers-list {
  font-family: "Ubuntu";
}
.Developers-loader-container {
  margin-top: -15rem;
}

.Developers-loader {
  max-width: 150px;
}

.btn.btn-red {
  background-color: #e54f4f;
  padding: 0.3rem 0.5rem;
  border-radius: 5px;
  font-size: 0.9rem;
  font-family: "Ubuntu";
  margin: 0 0.25rem;
  border: 1px solid #e54f4f;
  color: #fff;
}

.text-primary {
  color: #e54f4f !important;
}

.bg-primary {
  background-color: #e54f4f !important;
}

.btn-primary {
  color: #fff;
  background-color: #e54f4f;
  border-color: #e54f4f;
}

h4 {
  font-size: 1rem;
}

.post .round-img {
  max-width: 75px;
}

.post {
  display: grid;
  grid-template-columns: 1fr 7fr;
  grid-gap: 0rem;
  align-items: center;
}

a {
  color: #e54f4f;
  text-decoration: none;
  background-color: transparent;
}

.badge-primary {
  color: #fff;
  background-color: #e54f4f;
}

.center {
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  text-align: center;
}

.post {
  display: grid;
  grid-template-columns: 1fr 7fr;
  grid-gap: 0rem;
  align-items: center;
  text-align: start;
}
