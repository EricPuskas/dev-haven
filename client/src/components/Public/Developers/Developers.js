import React, { useEffect } from "react";
import { connect } from "react-redux";
import Loader from "../../../assets/img/rolling.svg";
import { getProfiles } from "../../../actions/profile";
import Developer from "./Developer";
import "./Developers.css";

const Developers = ({ getProfiles, profile: { profiles, loading } }) => {
  useEffect(() => {
    getProfiles();
  }, [getProfiles]);
  return (
    <div className="Developers">
      {loading ? (
        <div className="Developers-loader-container">
          <img src={Loader} alt="Loading..." className="Developers-loader" />
        </div>
      ) : (
        <>
          <h1>Developers</h1>
          <p>Browse and connect with Developers</p>
          <div className="Developers-list">
            {profiles.length > 0 ? (
              profiles.map(profile => (
                <Developer key={profile._id} profile={profile} />
              ))
            ) : (
              <h4>No profiles found</h4>
            )}
          </div>
        </>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  profile: state.profile
});

export default connect(
  mapStateToProps,
  { getProfiles }
)(Developers);
