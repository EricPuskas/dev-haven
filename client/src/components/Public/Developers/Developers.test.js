import { shallow } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Developers from "./Developers";

const mockStore = configureMockStore();
const store = mockStore({});
const mockProps = {
  getProfiles: jest.fn(),
  profile: {
    profiles: [
      {
        user: {
          _id: 1,
          name: "John",
          avatar: "Something"
        },
        status: "Dev",
        company: "Facebook",
        location: "USA",
        skills: [1, 2, 3, 4, 5]
      }
    ]
  },
  loading: false
};
const wrapper_filled = shallow(
  <Provider store={store}>
    <Developers {...mockProps} />
  </Provider>
);
describe("Test Developers Component", () => {
  it("renders component without crash", () => {
    expect(wrapper_filled.debug()).toMatchSnapshot();
  });
  const mockPropsNoProfiles = {
    getProfiles: jest.fn(),
    profile: {
      profiles: []
    },
    loading: false
  };
  const wrapper = shallow(
    <Provider store={store}>
      <Developers {...mockPropsNoProfiles} />
    </Provider>
  );
  const profileArr = wrapper.props().children.props.profile.profiles;
  it("expects no profiles", () => {
    expect(profileArr.length).toEqual(0);
  });
  it("show No profiles if empty", () => {
    expect(wrapper.find("#no-profiles-found").exists()).toBe(false);
  });
});
