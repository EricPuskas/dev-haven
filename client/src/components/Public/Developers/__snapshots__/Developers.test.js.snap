// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Developers Component renders component without crash 1`] = `
"<ContextProvider value={{...}}>
  <Connect(Developers) getProfiles={[Function: mockConstructor]} profile={{...}} loading={false} />
</ContextProvider>"
`;
